tests/unit/dataactvalidator/test_d41_detached_award_financial_assistance_5.py

```diff
@@ -0,0 +1,78 @@
+from tests.unit.dataactcore.factories.staging import DetachedAwardFinancialAssistanceFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+from dataactcore.models.domainModels import Zips
+
+_FILE = 'd41_detached_award_financial_assistance_5'
+
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "place_of_performance_code", "place_of_performance_zip4a"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset == actual
+
+
+def test_success(database):
+    """ The provided PrimaryPlaceofPerformanceZIP+4 must be in the state specified by PrimaryPlaceOfPerformanceCode.
+        In this specific submission row, the first five digits are valid and located in the correct state, but the
+        last 4 are invalid."""
+
+    # XX00000 validates here because it passes as long as the zip is valid in that state, this is checked
+    # in a different place
+    zips = Zips(zip5="12345", zip_last4="6789", state_abbreviation="NY")
+    # ignored because no zip4
+    det_award_1 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="NY*****",
+                                                          place_of_performance_zip4a="")
+    det_award_2 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="Ny**123",
+                                                          place_of_performance_zip4a=None)
+    # valid 5 digit zip
+    det_award_3 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="Ny**123",
+                                                          place_of_performance_zip4a="12345")
+    det_award_4 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="NY98765",
```

Comment:
Do we need 2 tests for a valid 5-digit zip if it's being ignored in this rule? You already have a 5-digit zip section and if invalid ones are being ignored we can pretty much assume valid ones are passing (although you could move one of these down there just in case)